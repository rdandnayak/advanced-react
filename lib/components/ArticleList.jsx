import React from 'react';
import Article from './Article';

const ArticleList = props => {
  return (
    <section>
      {Object.values(props.articles).map(article => {
        return <Article key={article.id} article={article} />;
      })}
    </section>
  );
};

export default ArticleList;
