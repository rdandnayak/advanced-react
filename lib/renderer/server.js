import React from 'react';
import ReactDOMServer from 'react-dom/Server';

import App from 'components/App';

const ServerRender = () => {
  return ReactDOMServer.renderToString(<App />);
};

export default ServerRender;
