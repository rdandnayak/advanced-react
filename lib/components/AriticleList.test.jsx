import React from 'react';
import ArticleList from './ArticleList';
import renderer from 'react-test-renderer';

describe('<ArticleList />', () => {
  it('renders correctly', () => {
    const testProps = {
      articles: {
        a: { id: 'a' },
        b: { id: 'b' }
      },
      articleActions: {
        lookupAuthor: jest.fn(() => ({}))
      }
    };
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
