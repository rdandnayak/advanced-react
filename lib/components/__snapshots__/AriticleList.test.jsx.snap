// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> renders correctly 1`] = `
<section>
  <section
    styles={undefined}
  >
    <div
      style={
        Object {
          "backgroundColor": "#ccc",
          "borderBottomColor": "#ccc",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 10,
          "paddingBottom": 10,
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.75em",
        }
      }
    >
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
        style={
          Object {
            "display": "inline-block",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 20,
        }
      }
    />
  </section>
  <section
    styles={undefined}
  >
    <div
      style={
        Object {
          "backgroundColor": "#ccc",
          "borderBottomColor": "#ccc",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 10,
          "paddingBottom": 10,
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.75em",
        }
      }
    >
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
        style={
          Object {
            "display": "inline-block",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingLeft": 20,
        }
      }
    />
  </section>
</section>
`;
